veps/sig-compute/10-dra-devices/vep.md

```diff
@@ -0,0 +1,781 @@
+	// this fields requires DRA feature gate enabled
+	ClaimRequest *ClaimRequest `json:",inline,omitempty"`
+}
+
+type ClaimRequest struct {
+	// ClaimName needs to be provided from the list vmi.spec.resourceClaims[].name where this
+	// device is allocated
+	ClaimName string `json:"claimName"`
+	// RequestName needs to be provided from resourceClaim.spec.devices.requests[].name where this
+	// device is requested
+	RequestName string `json:"requestName"`
+}
+
+type VirtualMachineInstanceStatus struct {
+	..
+	..
+	// DeviceStatus reflects the state of devices requested in spec.domain.devices.
+	// This is an optional field available only when DRA feature gate is enabled
+	// +optional
+	DeviceStatus *DeviceStatus `json:"deviceStatus,omitempty"`
+}
+
+// DeviceStatus has the information of all devices allocated spec.domain.devices
+type DeviceStatus struct {
+	// GPUStatuses reflects the state of GPUs requested in spec.domain.devices.gpus
+	// +listType=atomic
+	// +optional
+	GPUStatuses []DeviceStatusInfo `json:"gpuStatuses,omitempty"`
+	// HostDeviceStatuses reflects the state of GPUs requested in spec.domain.devices.hostDevices
+	// +listType=atomic
+	// +optional
+	HostDeviceStatuses []DeviceStatusInfo `json:"hostDeviceStatuses,omitempty"`
+}
+
+type DeviceStatusInfo struct {
+	// Name of the device as specified in spec.domain.devices.gpus.name or spec.domain.devices.hostDevices.name
+	Name string `json:"name"`
+	// DeviceResourceClaimStatus reflects the DRA related information for the device
+	DeviceResourceClaimStatus *DeviceResourceClaimStatus `json:"deviceResourceClaimStatus,omitempty"`
+}
+
+type DeviceResourceClaimStatus struct {
+	// Name is the name of actual device on the host provisioned by the driver as reflected in
+	// resourceclaim.status
+	// +optional
+	Name string `json:"name"`
+	// Attributes are the attributes for the allocated device. This information is published by the driver
+	// running on the node in resourceslice.spec.devices.basic.attributes for the allocated device.
+	// +optional
+	Attributes *DeviceAttributes `json:"attributes,omitempty"`
+	// ResourceClaimName is the name of the resource claim object used to provision this resource
+	// +optional
+	ResourceClaimName *string `json:"resourceClaimName,omitempty"`
+}
+
+type DeviceAttributes struct {
+	// PCIAddress is the PCIe bus address of the allocated device
+	// +optional
+	PCIAddress *string `json:"pciAddress,omitempty"`
+	// MDevUUID is the mediated device uuid of the allocated device
+	// +optional
+	MDevUUID *string `json:"mdevUUID,omitempty"`
+}
+```
+
+The first section vmi.spec.resourceClaims will have a list of devices needed to be allocated for the VM. Having this
+available as a list will allow users to use the device from this list in GPU section or HostDevices section of the
+DomainSpec API.
+
+In v1alpha3 version of [DRA API](https://pkg.go.dev/k8s.io/api@v0.31.0/resource/v1alpha3#DeviceClaim), multiple drivers
```

Comment:
We should build this on the beta that is present now as far I see

Reply:
+1, will update to beta.

Reply:
updated